pkg/server/apiHandler.go

```diff
@@ -301,6 +304,65 @@ func (h *apiHandler) logs(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusCreated)
 }
 
+func (h *apiHandler) newLogs(w http.ResponseWriter, r *http.Request) {
+	cert := getClientCert(r)
+	u, err := h.manager.DeviceCheckCert(cert)
+	if err != nil {
+		log.Printf("error checking device cert: %v", err)
+		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+		return
+	}
+	h.recordClient(u, r)
+	gr, err := gzip.NewReader(r.Body)
```

Comment:
need to add something like, since the newlog message is only sent in APIV2:
	sm := &zauth.AuthContainer{}
	err := proto.Unmarshal(c, sm)
	data := sm.ProtectedPayload.GetPayload()

Reply:
It works fine without any modifications for APIV1, and I checked it. Actually, AuthContainer logic depends on variable [inside ZedCloudContext](https://github.com/naiming-zededa/eve/blob/f5db03d68191618031c3b3b993f5ba8784667a36/pkg/pillar/zedcloud/send.go#L399). You build the correct url for device logs [here](https://github.com/naiming-zededa/eve/blob/f5db03d68191618031c3b3b993f5ba8784667a36/pkg/pillar/cmd/loguploader/loguploader.go#L355). 
But I cannot figure out what is stopping you from creating the correct link for application logs (pass `zedcloudCtx.V2API` instead of `true` [here](https://github.com/naiming-zededa/eve/blob/naiming-newlog-infra/pkg/pillar/cmd/loguploader/loguploader.go#L580-L581) and modify [relative link](https://github.com/lf-edge/eve/pull/1667#discussion_r531123817)).

Reply:
ok, let me change the url formatting to support the APIV1 correctly. Didn't realize adam was using that still.